Makefile

```diff
@@ -335,9 +335,25 @@ prebuilt-binary:
 		-v `pwd`:/go/src/$(PACKAGE_NAME) \
 		-w /go/src/$(PACKAGE_NAME) \
 		ghcr.io/goreleaser/goreleaser-cross:${GOLANG_CROSS_VERSION} \
-		release --clean --parallelism 1
+		release --clean --parallelism 2
 .PHONY: prebuilt-binary
 
+## go-releaser-dry-run ensures that the go releaser tool and build all the artefacts correctly.
+## specifies parallelism as 4 so should be run locally. On the regular github runners 2 should be the max.
+go-releaser-dry-run:
+	@echo "Running GoReleaser in dry-run mode..."
+	docker run \
+		--rm \
+		--env CGO_ENABLED=1 \
+		--env GORELEASER_CURRENT_TAG=${GORELEASER_CURRENT_TAG} \
+		--env-file .release-env \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v `pwd`:/go/src/$(PACKAGE_NAME) \
+		-w /go/src/$(PACKAGE_NAME) \
+		ghcr.io/goreleaser/goreleaser-cross:${GOLANG_CROSS_VERSION} \
+		release --snapshot --clean --parallelism 4
```

Comment:
the `--snapshot` disables publishing. I can refactor this to not have as much duplication if it is preferred.